src/mvba/bundle.rs

```diff
@@ -11,7 +11,14 @@ pub struct Bundle {
     /// This is the destination  module, it can be ABBA, VCBC or MVBA.
     pub module: String,
     /// This is the actual message
-    pub payload: Vec<u8>,
+    pub(crate) message: Message,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq)]
+pub enum Message {
+    VcbcMsg(vcbc::message::Message),
+    AbbaMsg(abba::message::Message),
+    MvbaMsg(mvba::message::Message),
 }
```

Comment:
The `*Msg` suffix is redundant. They are already prefixed by `Message::*`
```suggestion
pub enum Message {
    Vcbc(vcbc::message::Message),
    Abba(abba::message::Message),
    Mvba(mvba::message::Message),
}
```